ipatests/pytest_ipa/integration/__init__.py

```diff
@@ -386,45 +386,78 @@ def mh(request, class_integration_logs):
     """
     cls = request.cls
 
-    domain_description = {
-        'type': 'IPA',
-        'hosts': {
-            'master': 1,
-            'replica': cls.num_replicas,
-            'client': cls.num_clients,
-        },
-    }
-    domain_description['hosts'].update(
-        {role: 1 for role in cls.required_extra_roles})
-
-    domain_descriptions = [domain_description]
-    for _i in range(cls.num_ad_domains):
-        domain_descriptions.append({
-            'type': 'AD',
-            'hosts': {'ad': 1}
-        })
-    for _i in range(cls.num_ad_subdomains):
-        domain_descriptions.append({
-            'type': 'AD_SUBDOMAIN',
-            'hosts': {'ad_subdomain': 1}
-        })
-    for _i in range(cls.num_ad_treedomains):
-        domain_descriptions.append({
-            'type': 'AD_TREEDOMAIN',
-            'hosts': {'ad_treedomain': 1}
-        })
-
-    mh = make_multihost_fixture(
-        request,
-        domain_descriptions,
-        config_class=Config,
-        _config=get_global_config(),
-    )
+    if cls.num_trusted_domains == 1:
+        mh = make_multihost_fixture(request, descriptions=[
+            {
+                'type': 'IPA',
+                'hosts':
+                {
+                    'master': 1,
+                    'replica': cls.num_replicas,
+                    'client': cls.num_clients,
+                }
+            },
+            {
+                'type': 'TRUSTED_IPA',
+                'hosts':
+                {
+                    'master': 1,
+                    'replica': cls.num_trusted_replicas,
+                    'client': cls.num_trusted_clients,
+
+                },
+            },
+        ], config_class=Config, _config=get_global_config(),)
+    else:
+        domain_description = {
+            'type': 'IPA',
+            'hosts': {
+                'master': 1,
+                'replica': cls.num_replicas,
+                'client': cls.num_clients,
+            },
+        }
+        domain_description['hosts'].update(
+            {role: 1 for role in cls.required_extra_roles})
+
+        domain_descriptions = [domain_description]
+        for _i in range(cls.num_ad_domains):
+            domain_descriptions.append({
+                'type': 'AD',
+                'hosts': {'ad': 1}
+            })
+        for _i in range(cls.num_ad_subdomains):
+            domain_descriptions.append({
+                'type': 'AD_SUBDOMAIN',
+                'hosts': {'ad_subdomain': 1}
+            })
+        for _i in range(cls.num_ad_treedomains):
+            domain_descriptions.append({
+                'type': 'AD_TREEDOMAIN',
+                'hosts': {'ad_treedomain': 1}
+            })
+        mh = make_multihost_fixture(
+            request,
+            domain_descriptions,
+            config_class=Config,
+            _config=get_global_config(),
+        )
+    if cls.num_trusted_domains == 1:
```

Comment:
Instead of having 2 separate code branches you can keep a single one and add
```
    for domain in mh.config.domains:
         if domain.type == 'IPA':
              mh.domain = domain
         elif domain.type == 'TRUSTED_IPA':
              mh.trusted_domain = domain

    [mh.master] = mh.domain.hosts_by_role('master')
    mh.replicas = mh.domain.hosts_by_role('replica')
    mh.clients = mh.domain.hosts_by_role('client')
    if mh.config.trusted_domains:
        [mh.trusted_master] = mh.trusted_domain.hosts_by_role('master')
        mh.trusted_replicas = mh.trusted_domain.hosts_by_role('replica')
        mh.trusted_clients = mh.trusted_domain.hosts_by_role('client')

```